### PR TITLE
Fix: Update WebSocket URL to use correct protocol from window.location

### PR DESCRIPTION
**Pull request by SmartManoj**
Solves https://github.com/OpenDevin/OpenDevin/issues/1649#issuecomment-2107239803

### CODE REVIEW

File: frontend/src/services/socket.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ class Socket {
   private static _initialize(token: string): void {
     if (Socket.isConnected()) return;
 
-    const WS_URL = `ws://${window.location.host}/ws?token=${token}`;
+    const WS_URL = `${window.location.protocol}//${window.location.host}/ws?token=${token}`;
```
Comment (isavita) [inline frontend/src/services/socket.ts:44]: Shouldn't it be something like
```bash
const protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:';
const WS_URL = `${protocol}//${window.location.host}/ws?token=${token}`;
```

Comment (SmartManoj) [reply frontend/src/services/socket.ts:44]: Yes

Comment (SmartManoj) [reply frontend/src/services/socket.ts]: https://haacked.com/archive/2019/06/03/suggested-changes/

**Review by isavita — APPROVED** [review]

**Comment by iFurySt** [comment]
https://github.com/OpenDevin/OpenDevin/pull/1248#issuecomment-2119013483

@SmartManoj @isavita @amanape do you have any ideas to solve this question? i think the two questions are related

**Comment by SmartManoj** [comment]
@iFurySt, This is the hardcoded value mentioned in that PR. It will solve those browser errors.

**Comment by amanape** [comment]
Yes I believe this update should resolve both issues. A lot of annoying things happen when trying to create insecure connections with ws to https.

**Review by amanape — APPROVED** [review]

**Pull request merged**